// react-navbar/build/src/MenuItems.js
'use strict';

Object.defineProperty(exports, '__esModule', {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactIntl = require('react-intl');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Item = require('./Item');

var _Item2 = _interopRequireDefault(_Item);

var _Dropdown = require('./Dropdown');

var _Dropdown2 = _interopRequireDefault(_Dropdown);

exports['default'] = _react2['default'].createClass({
    displayName: 'MenuItems',

    propTypes: {
        items: _react2['default'].PropTypes.array.isRequired,
        secondary: _react2['default'].PropTypes.bool
    },

    mixins: [_reactIntl.IntlMixin],

    render: function render() {
        var _this = this;

        var className = (0, _classnames2['default'])('nav navbar-nav', { 'navbar-right': this.props.secondary });

        var items = this.props.items.map(function (item, index) {
            return _this.itemElement(item, 'i' + index);
        });

        return _react2['default'].createElement(
            'ul',
            { className: className },
            items
        );
    },

    itemElement: function itemElement(item, ref) {
        return item.items ? _react2['default'].createElement(_Dropdown2['default'], _extends({}, item, { ref: ref, key: ref })) : _react2['default'].createElement(_Item2['default'], _extends({}, item, { ref: ref, key: ref }));
    }
});
module.exports = exports['default'];
