// react-navbar/build/src/Navbar.js
'use strict';

Object.defineProperty(exports, '__esModule', {
    value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactIntl = require('react-intl');

var _MenuItems = require('./MenuItems');

var _MenuItems2 = _interopRequireDefault(_MenuItems);

exports['default'] = _react2['default'].createClass({
    displayName: 'Navbar',

    propTypes: {
        menuItems: _react2['default'].PropTypes.array.isRequired,
        secondaryMenuItems: _react2['default'].PropTypes.array
    },

    mixins: [_reactIntl.IntlMixin],

    render: function render() {
        var secondaryMenuItems = this.props.secondaryMenuItems ? _react2['default'].createElement(_MenuItems2['default'], {
            ref: 'secondaryMenuItems',
            secondary: true,
            items: this.props.secondaryMenuItems }) : null;

        return _react2['default'].createElement(
            'nav',
            { className: 'navbar navbar-default' },
            _react2['default'].createElement(
                'div',
                { className: 'container' },
                _react2['default'].createElement(
                    'div',
                    { className: 'navbar-header' },
                    _react2['default'].createElement(
                        'button',
                        { type: 'button', className: 'navbar-toggle collapsed', 'data-toggle': 'collapse', 'data-target': '#react-navbar-collapse-all', 'aria-expanded': 'false' },
                        _react2['default'].createElement(
                            'span',
                            { className: 'sr-only' },
                            'Toggle navigation'
                        ),
                        _react2['default'].createElement('span', { className: 'icon-bar' }),
                        _react2['default'].createElement('span', { className: 'icon-bar' }),
                        _react2['default'].createElement('span', { className: 'icon-bar' })
                    )
                ),
                _react2['default'].createElement(
                    'div',
                    { className: 'collapse navbar-collapse', id: 'react-navbar-collapse-all' },
                    _react2['default'].createElement(_MenuItems2['default'], { ref: 'menuItems', items: this.props.menuItems }),
                    secondaryMenuItems
                )
            )
        );
    }
});
module.exports = exports['default'];
